src/DynamoPublish/ShareWorkspaceClient.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Greg;
+using Greg.Requests;
+using Greg.Responses;
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dynamo.Publish
+{
+    public class ShareWorkspaceClient
+    {
+        private IAuthProvider _authenticationProvider;
+        private string _baseUrl;
+        private IGregClient _client;
+
+        public IAuthProvider AuthenticationProvider
+        {
+            get { return _authenticationProvider; }
+        }
+
+        public string BaseUrl
+        {
+            get { return _baseUrl; }
+        }
+
+        public ShareWorkspaceClient(string baseUrl, IAuthProvider authProvider)
+        {
+            _baseUrl = baseUrl;
+            _authenticationProvider = authProvider;
+            _client = new GregClient(_authenticationProvider, _baseUrl);
```

Comment:
@pbidenko Let's not use `GregClient` here. Although it works (right now), it exposes this code to potential regressions associated with a completely different part of the app. We should add this capability to `Reach` in the `WorkspaceStorageClient`. It looks repetitive, but the code is ultimately so simple that this should not matter.